qa/tasks/cephfs/test_volumes.py

```diff
@@ -4251,6 +4253,218 @@ def test_subvolume_rm_with_snapshots(self):
+        self._do_subvolume_io(sv, number_of_files=100)
+        sv_files = self._note_files_in_sv(sv_path)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        self._assert_trashed_sv_is_unpurged(sv, sv_path, sv_files)
+        time.sleep(7)
+        self._assert_trashed_sv_is_unpurged(sv, sv_path, sv_files)
+
+        # XXX actual test here: test that unpurged subvol is purged
+        self.run_ceph_cmd(f'config set mgr {self.conf_opt} false')
+        self.wait_for_trashcan_to_be_empty()
+
+    def _assert_trashed_sv_not_fully_purged_yet(self, sv, sv_path):
+        uuid = self.mount_a.get_shell_stdout('sudo ls volumes/_deleting').\
+                    strip()
+
+        trash_sv_path = sv_path.replace('_nogroup', f'_deleting/{uuid}')
+        trash_sv_path = trash_sv_path.replace(sv, '')
+
+        sv_files= self.mount_a.get_shell_stdout(f'sudo ls {trash_sv_path}').\
+            strip().split('\n')
+        self.assertGreater(len(sv_files),  0)
+
+    def test_when_changed_to_true_ongoing_purge_halts(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging_trash is
+        set to true, running "ceph fs subvolume rm" will move the subvolume
+        to trash but (asynchronous) purge of the subvolume won't happen till
+        this config option is not explicitly set to False.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self.run_ceph_cmd(f'config set mgr {self.conf_opt} false')
+
+        sv_path = self._get_sv_path(v, sv)
+        # adding more files for this test since in once few runs it fails due
+        # to race condition
+        self._do_subvolume_io(sv, number_of_files=500)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        # XXX actual test here: test that purging halts
+        self.run_ceph_cmd(f'config set mgr {self.conf_opt} true')
+        self._assert_trashed_sv_not_fully_purged_yet(sv, sv_path)
+        time.sleep(7)
+        self._assert_trashed_sv_not_fully_purged_yet(sv, sv_path)
+
+    def test_when_changed_to_false_partly_purged_subvol_purges_fully(self):
+        '''
+        Test that when MGR config option mgr/volumes/pause_purging_trash is
+        changed to false, the async purging of a subvolume will resume and
+        also finish, causing trash to be empty.
+        '''
+        v = self.volname
+        sv = 'sv1'
+
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv} --mode=777')
+        self._do_subvolume_io(sv, number_of_files=100)
+
+        self.run_ceph_cmd(f'fs subvolume rm {v} {sv}')
+        self.run_ceph_cmd(f'config set mgr {self.conf_opt} true')
+        time.sleep(2)
+
+        # XXX actual test here: test that purging is resumed and finished
+        self.run_ceph_cmd(f'config set mgr {self.conf_opt} false')
+        self.wait_for_trashcan_to_be_empty()
+
+    def test_the_default_value(self):
```

Comment:
Does this really needs to be a test?

Reply:
Yes. If this gets messed up, subvolumes won't be purged by default which will qualify as a bug introduced by this PR.

Reply:
Messed up how? You mean some code that gets merged? (that code would be vetted during review)

Reply:
> that code would be vetted during review

Yeah but it's possible to miss and one small test won't hurt.   :)

Reply:
That does not make it a candidate for a test.